### PR TITLE
BTAT-2770: Added FinancialTransactionsModel with associated API reads

### PR DESCRIPTION
**Pull request by chrisjwwalker**

### CODE REVIEW

File: app/models/viewModels/FinancialTransactionsModel.scala [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+      identity
+    )
+  }
+
+  implicit val reads: Reads[Seq[FinancialTransactionsModel]] = new Reads[Seq[FinancialTransactionsModel]] {
+    override def reads(json: JsValue): JsResult[Seq[FinancialTransactionsModel]] = {
+      val transactionsList = json.get[List[JsValue]]("financialTransactions")
+
+      def getItemsForPeriod(transaction: JsValue): List[(Long, LocalDate)] = {
+        transaction.\("items").validate[List[JsValue]].fold(
```
Comment (TomDudley92) [inline app/models/viewModels/FinancialTransactionsModel.scala]: Can you filter for charge type of VAT Return charge as well please

File: test/models/FinancialTransactionsModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+
+import java.time.LocalDate
+
+import models.viewModels.FinancialTransactionsModel
+import play.api.libs.json.{JsSuccess, Json}
+import uk.gov.hmrc.play.test.UnitSpec
+
+class FinancialTransactionsModelSpec extends UnitSpec {
+
+  val reads = FinancialTransactionsModel.reads
```
Comment (TomDudley92) [inline test/models/FinancialTransactionsModelSpec.scala]: Can you add a type annotation please

File: app/models/viewModels/FinancialTransactionsModel.scala [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.time.LocalDate
+
+import play.api.libs.json._
+
+case class FinancialTransactionsModel(taxPeriodFrom: LocalDate,
+                                      taxPeriodTo: LocalDate,
+                                      amount: Long,
+                                      clearedDate: LocalDate)
+
+object FinancialTransactionsModel {
```
[the remark this answers is not in the document]
Comment (chrisjwwalker) [reply app/models/viewModels/FinancialTransactionsModel.scala]: 👌🏼

File: test/models/FinancialTransactionsModelSpec.scala (hunk 2 above)
Comment (chrisjwwalker) [reply test/models/FinancialTransactionsModelSpec.scala]: 👌🏼

**Comment by chrisjwwalker** [comment]
Renamed the model, added the type annotion and added the filter by `chargeType`

**Pull request merged**